src/ebml.rs

```diff
@@ -264,24 +265,23 @@ pub struct EBMLHeader {
 pub fn ebml_header(input: &[u8]) -> IResult<&[u8], EBMLHeader, Error> {
     ebml_master(0x1A45DFA3, |i| {
         map(
-            permutation((
-                ebml_uint(0x4286), // version
-                ebml_uint(0x42F7), // read_version
-                ebml_uint(0x42F2), // max id length
-                ebml_uint(0x42F3), // max size length
-                ebml_str(0x4282),  // doctype
-                ebml_uint(0x4287), // doctype version
-                ebml_uint(0x4285), // doctype_read version
-                opt(complete(skip_void)),
+            matroska_permutation((
+                complete(ebml_uint(0x4286)), // version
+                complete(ebml_uint(0x42F7)), // read_version
+                complete(ebml_uint(0x42F2)), // max id length
+                complete(ebml_uint(0x42F3)), // max size length
+                complete(ebml_str(0x4282)),  // doctype
+                complete(ebml_uint(0x4287)), // doctype version
+                complete(ebml_uint(0x4285)), // doctype_read version
             )),
             |t| EBMLHeader {
-                version: t.0,
-                read_version: t.1,
-                max_id_length: t.2,
-                max_size_length: t.3,
-                doc_type: t.4,
-                doc_type_version: t.5,
-                doc_type_read_version: t.6,
+                version: t.0.unwrap(),
```

Comment:
Why all the unwraps now? Can we avoid that? :) Either things are optional or they're not, but if they're not optional then the parser should fail cleanly instead of just panicking.

Reply:
I tried to implement that, but with the `unwrap` I can precisely see which struct fields are failing. Returning an error does not show where the parser is failing. Do you have another solution in mind? For example an error function like the `usize_error` one?

Reply:
https://github.com/zesterer/ariadne might be of help?

Reply:
An `unwrap()` is very disruptive while a parser error can easily be handled by the caller. You can return a custom error here in one way or another, but panicking doesn't seem very nice.

This article also has some example how to do nice errors with nom: https://fasterthanli.me/series/making-our-own-ping/part-9

I would keep "make errors nice" for a separate PR though.

----

Generally, you should never panic on broken files :)

Reply:
Ok, let me try to remove the `unwrap` in some way. The custom error option is the most suitable in my opinion at the very moment. Thanks for the help! :)

@lu-zero I wanted to avoid adding more dependencies if it is possible :)